### PR TITLE
Python build updates

### PR DESCRIPTION
**Pull request by jgriffiths**
From investigating https://github.com/Blockstream/gdk/issues/67 - Also fixes OSX Travis build.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
       osx_image: xcode11.3
       install:
-        - brew update
-        - brew upgrade
-        - brew install ninja autoconf libtool gnu-sed python3 wget pkg-config swig gnu-getopt gnu-tar
         - python3 -m pip install virtualenv meson===0.51.2
         - curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain 1.42.0
       script:
-        - source $HOME/.cargo/env && ./tools/build.sh --clang --python-version 3.7 --enable-rust
+        - source $HOME/.cargo/env && ./tools/build.sh --clang --python-version $(python3 --version | sed -e 's/.* //' -e 's/.[0-9]$//') --enable-rust
```
Comment (LeoComandini) [inline .travis.yml:25]: This does not work if the patch version is > 9

    echo "Python 3.6.10" | sed -e 's/.* //' -e 's/.[0-9]$//'
    3.6.

The regex should be tweaked or we can use

     python3 -c "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')"

Comment (jgriffiths) [reply .travis.yml:25]: Tweaked, thanks

**Comment by LeoComandini** [comment]
utACK 20ac97a1a36b70d9f9596820788f88c6ddec7fdc

Assuming the regex fix is pushed.
Also, before merging, we might want to have a mac user to do a smoketest with a py3.8 wheel.

File: .travis.yml [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
       osx_image: xcode11.3
       install:
-        - brew update
-        - brew upgrade
-        - brew install ninja autoconf libtool gnu-sed python3 wget pkg-config swig gnu-getopt gnu-tar
         - python3 -m pip install virtualenv meson===0.51.2
         - curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain 1.42.0
       script:
-        - source $HOME/.cargo/env && ./tools/build.sh --clang --python-version 3.7 --enable-rust
+        - source $HOME/.cargo/env && ./tools/build.sh --clang --python-version $(python3 -c "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')") --enable-rust
```
Comment (jkauffman1) [inline .travis.yml:25]: If we passed `--python-version 3` here, and thus `python3` to pymodule.find_installation in meson.build, it looks like meson should automatically discover the right python?

Comment (jgriffiths) [reply .travis.yml:25]: meson couldn't discover its own shoes, sadly.

There can be multiple python3 versions installed, and versions 3.8+ require a hack that previous versions don't. So I think we have to keep this for now (or at least until meson is updated/we update, which no doubt will come with its own new problems).

**Comment by jkauffman1** [comment]
utack

**Comment by jgriffiths** [comment]
merged, closing

**Pull request closed without merging**